federation/pkg/federation-controller/namespace/namespace_controller.go

```diff
@@ -371,6 +416,13 @@ func (nc *NamespaceController) delete(namespace *api_v1.Namespace) error {
 		return fmt.Errorf("failed to delete events list from namespace: %v", err)
 	}
 
+	// Delete the namespace from all underlying clusters.
+	updatedNamespaceObj, err := nc.deletionHelper.ProcessDeletion(updatedNamespace)
```

Comment:
I thought ProcessDeletion should take a `fed_v1.Namespace`, why is it taking a regular namespace?